tools/entreprise-provisionning/src/Types.hs

```diff
@@ -0,0 +1,119 @@
+-- This file is part of the Wire Server implementation.
+--
+-- Copyright (C) 2025 Wire Swiss GmbH <opensource@wire.com>
+--
+-- This program is free software: you can redistribute it and/or modify it under
+-- the terms of the GNU Affero General Public License as published by the Free
+-- Software Foundation, either version 3 of the License, or (at your option) any
+-- later version.
+--
+-- This program is distributed in the hope that it will be useful, but WITHOUT
+-- ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+-- FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more
+-- details.
+--
+-- You should have received a copy of the GNU Affero General Public License along
+-- with this program. If not, see <https://www.gnu.org/licenses/>.
+
+module Types where
+
+import Data.Aeson
+import Data.Id
+import Data.Range
+import GHC.Generics
+import Imports
+
+newtype Token = Token {fromToken :: Text}
+  deriving stock (Eq, Show, Generic)
+
+newtype ApiUrl = ApiUrl {fromApiUrl :: String}
+  deriving stock (Eq, Generic)
+  deriving newtype (Show)
+
+data Services = Services
+  { galleyUrl :: ApiUrl,
+    brigUrl :: ApiUrl
+  }
+  deriving stock (Eq, Show, Generic)
+
+newtype ChannelName = ChannelName {fromChannelName :: Range 1 256 Text}
+  deriving stock (Eq, Show, Generic)
+  deriving newtype (ToJSON, FromJSON)
+
+newtype UserGroupChannelsProvisionningSpec = UserGroupChannelsProvisionningSpec
+  { byGroup :: Map UserGroupId [ChannelName]
+  }
+  deriving stock (Eq, Show, Generic)
+  deriving newtype (ToJSON, FromJSON)
+
+newtype UserGroupChannelsProvisionningResult = UserGroupChannelsProvisionningResult
+  { results :: Map UserGroupId UserGroupResult
+  }
+  deriving stock (Eq, Show, Generic)
+  deriving newtype (FromJSON, ToJSON)
+
+data UserGroupResult = UserGroupResult
+  { channel :: [ChannelResult],
+    association :: AssociationResult
+  }
+  deriving stock (Eq, Show, Generic)
+  deriving (FromJSON, ToJSON) via (Generically UserGroupResult)
+
+data ChannelResult
+  = ChannelSuccess ChannelName ConvId
+  | ChannelFailure ChannelName ErrorDetail
+  deriving stock (Eq, Show, Generic)
+
+channelResultName :: ChannelResult -> ChannelName
+channelResultName (ChannelSuccess n _) = n
+channelResultName (ChannelFailure n _) = n
+
+instance ToJSON ChannelResult where
```

Comment:
please keep the aeson instance, but just checking: are you aware of schema-profunctor?  (in libs/ and services/ you should always use that for new code, and derive the aeson and openapi instances.)